### PR TITLE
Add unit tests for fine-grained join & agg

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: ref #6157

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by yibin87** [comment]
unhold until #6442 is fixed.

File: dbms/src/Flash/tests/gtest_compute_server.cpp [hunk 1]
```diff
@@ -445,5 +501,115 @@ try
...
+        case tipb::ExecType::TypeJoin:
+        {
+            /// update build side path
+            JoinInterpreterHelper::TiFlashJoin tiflash_join{current->join()};
+            current = const_cast<tipb::Executor *>(&current->join().children()[tiflash_join.build_side_index]);
+            break;
+        }
+        case tipb::ExecType::TypeExchangeSender:
+            if (current->exchange_sender().tp() == tipb::Hash)
+                current->set_fine_grained_shuffle_stream_count(8);
```
Comment (ywqzzy) [inline dbms/src/Flash/tests/gtest_compute_server.cpp:529]: We can modify the `AggregationBinder::toMPPSubPlan` to pass the `fine_grained_shuffle_stream_count` to `ExchangeReceiverBinder` and `ExchangeSenderBinder`.
Then the `ExchangexxxBinder` can use `fine_grained_shuffle_stream_count` to set the `tipb::Executor`

**Review by ywqzzy — COMMENTED** [review]
Rest LGTM

**Review by ywqzzy — APPROVED** [review]

File: dbms/src/Flash/tests/gtest_compute_server.cpp (hunk 1 above)
Comment (ywqzzy) [reply dbms/src/Flash/tests/gtest_compute_server.cpp:529]: > We can modify the `AggregationBinder::toMPPSubPlan` to pass the `fine_grained_shuffle_stream_count` to `ExchangeReceiverBinder` and `ExchangeSenderBinder`. Then the `ExchangexxxBinder` can use `fine_grained_shuffle_stream_count` to set the `tipb::Executor`

I will refactor it when this pr merged.

**Comment by SeaRise** [comment]
It looks like `ComputeServerRunner.runFineGrainedShuffleJoinTest` has had a crash.
```
[2022-12-23T04:54:59.670Z] /usr/local/bin/../include/c++/v1/vector:1580: _LIBCPP_ASSERT '__n < size()' failed. vector[] index out of bounds
```

**Comment by yibin87** [comment]
> It looks like `ComputeServerRunner.runFineGrainedShuffleJoinTest` has had a crash.
> 
> ```
> [2022-12-23T04:54:59.670Z] /usr/local/bin/../include/c++/v1/vector:1580: _LIBCPP_ASSERT '__n < size()' failed. vector[] index out of bounds
> ```

Looking at it, might introduced by rebase.

**Review by mengxin9014 — APPROVED** [review]
LGTM

**Comment by ywqzzy** [comment]
> > It looks like `ComputeServerRunner.runFineGrainedShuffleJoinTest` has had a crash.
> > ```
> > [2022-12-23T04:54:59.670Z] /usr/local/bin/../include/c++/v1/vector:1580: _LIBCPP_ASSERT '__n < size()' failed. vector[] index out of bounds
> > ```
> 
> Looking at it, might introduced by rebase.

Related issue: https://github.com/pingcap/tiflash/issues/6542

**Pull request merged**